### PR TITLE
Standardize AWS SQS classes names

### LINKED ISSUE #20296: Consistently name Amazon provider operators files
### Description

Follow-up to 
Restructure Amazon provider operators files [#20139](https://github.com/apache/airflow/issues/20139) as discussed with @eladkal 

Amazon Operators, Hooks, and Sensors are inconsistently named regarding acronym capitalization and whether to prefix with AWS or not.

An incomplete list for example: we have `SnsPublishOperator` but we capitalize the acronyms in `SQSPublishOperator` and `ECSOperator`. We prefix `AwsGlueCrawlerOperator` and `AwsBatchOperator` in lowercase, `AWSAthenaOperator` is prefixed in uppercase, and no prefix at all on `GlacierCreateJobOperator`or `CloudFormationCreateStackOperator`

Correct naming conventions are listed in AIP 21 cases [3-4](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-21%3A+Changes+in+import+paths#AIP21:Changesinimportpaths-Case#3{aws/azure/gcp}_*) and case [6](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-21%3A+Changes+in+import+paths#AIP21:Changesinimportpaths-Case#6Otherisolatedcases).

The sensors/operators class names should not have Aws/AWS as part of the class name and only the first letter of the acronyms should be capitalized.

So:
SnsPublishOperator -> No change
SQSPublishOperator -> SqsPublishOperator
AwsBatchOperator -> BatchOperator
AWSAthenaOperator -> AthenaOperator
etc...

### Use case/motivation

For each task: 
 -Rename the operator
 -Deprecate the old name
 --Add to tests/deprecated_classes.py
 --Add to the KNOWN_DEPRECATED_MESSAGES section of dev/provider_packages/prepare_provider_packages.py
 -Correct any references to the old name in docstrings and docs files
 -Correct all references in tests
 -Correct references in example dags

### Changes Required:

| Current                      | New Name                 |
|------------------------------|--------------------------|
| AWSAthenaOperator            | AthenaOperator           |
| AwsBatchOperator             | BatchOperator            |
| AWSDataSyncOperator          | DataSyncOperator         |
| ECSProtocol                  | EcsProtocol              |
| ECSTaskLogFetcher            | EcsTaskLogFetcher        |
| ECSOperator                  | EcsOperator              |
| EKS***Operator (all of them) | Eks***Operator           |
| SQSPublishOperator           | SqsPublishOperator       |
|||
| EKS***Sensor (all of them) | Eks***Sensor           |
| AwsRedshiftClusterSensor   | RedshiftClusterSensor  |
| SQSSensor                  | SqsSensor              |
|||
| AWSAthenaHook     | AthenaHook     |
| AWSCloudFormationHook     | CloudFormationHook     |
| AWSDataSyncHook     | DataSyncHook     |
| AwsDynamoDBHook     | DynamoDBHook     |
| EKSHook     | EksHook     |
| AwsFirehoseHook  (in kinesis.py)   | FirehoseHook     |
| AwsLambdaHook     | LambdaHook     |
| SESHook     | SesHook     |
| AwsSnsHook     | SnsHook     |
| SQSHook     | SqsHook     |

### Action Items:

- [x] operators/athena.py - @ferruzzi https://github.com/apache/airflow/pull/20305
- [x] operators/batch.py - @ferruzzi https://github.com/apache/airflow/pull/20369
- [x] operators/datasync.py - @ferruzzi https://github.com/apache/airflow/pull/20328
- [x] operators/ecs.py - @ferruzzi https://github.com/apache/airflow/pull/20332
- [x] operators/eks.py - @ferruzzi https://github.com/apache/airflow/pull/20354
- [x] operators/glue_crawler.py - @ferruzzi https://github.com/apache/airflow/pull/20372
- [x] operators/sqs.py - @eladkal https://github.com/apache/airflow/pull/20732
- [x] sensors/eks.py - @ferruzzi https://github.com/apache/airflow/pull/20354
- [x] sensors/redshift.py - @ferruzzi https://github.com/apache/airflow/pull/20374
- [x] sensors/sqs.py - @eladkal https://github.com/apache/airflow/pull/20732
- [x] hooks/athena.py - @ferruzzi https://github.com/apache/airflow/pull/20305
- [x] hooks/cloud_formation.py @ferruzzi https://github.com/apache/airflow/pull/20357
- [x] hooks/datasync.py - @ferruzzi https://github.com/apache/airflow/pull/20328
- [x] hooks/dynamodb.py - @ferruzzi  https://github.com/apache/airflow/pull/20360
- [x] hooks/emr.py - @bhavaniravi https://github.com/apache/airflow/pull/20375
- [x] hooks/eks.py - @ferruzzi https://github.com/apache/airflow/pull/20354
- [x] hooks/kinesis.py - @ferruzzi https://github.com/apache/airflow/pull/20362
- [x] hooks/lambda.py - @ferruzzi https://github.com/apache/airflow/pull/20365
- [x] hooks/ses.py - @ferruzzi https://github.com/apache/airflow/pull/20367
- [x] hooks/sns.py  - @ferruzzi https://github.com/apache/airflow/pull/20368
- [x] hooks/sqs.py - @eladkal https://github.com/apache/airflow/pull/20732

### To be verified (recently changed in the linked campaign):

~~DMS - good~~
~~EC2 - good~~
~~EMR - good~~
~~Glue - Added above~~
~~S3 - good~~
~~SageMaker - good~~

---

### No changes required:
**Operators:**
CloudFormationCreateStackOperator
CloudFormationDeleteStackOperator
GlacierCreateJobOperator
SnsPublishOperator
StepFunctionGetExecutionOutputOperator
StepFunctionStartExecutionOperator
AthenaSensor
CloudFormationCreateStackSensor
CloudFormationDeleteStackSensor
DmsTaskBaseSensor
DmsTaskCompletedSensor

**Sensors:**
GlacierJobOperationSensor
StepFunctionExecutionSensor

**Hooks:**
ElastiCacheReplicationGroupHook
GlacierHook
RedshiftHook
S3Hook
SageMakerHook
SecretsManagerHook
StepFunctionHook

### Related issues

https://github.com/apache/airflow/issues/20139

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by eladkal**
Final PR to close: https://github.com/apache/airflow/issues/20296

Replace https://github.com/apache/airflow/pull/20418

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by eladkal** [comment]
Ready for review :)
cc @ferruzzi

File: airflow/contrib/operators/aws_sqs_publish_operator.py [hunk 1]
```diff
@@ -19,10 +19,26 @@
...
 import warnings
 
-from airflow.providers.amazon.aws.operators.sqs import SQSPublishOperator  # noqa
+from airflow.providers.amazon.aws.operators.sqs import SqsPublishOperator
 
 warnings.warn(
     "This module is deprecated. Please use `airflow.providers.amazon.aws.operators.sqs`.",
     DeprecationWarning,
     stacklevel=2,
 )
```
Comment (ferruzzi) [inline airflow/contrib/operators/aws_sqs_publish_operator.py:28]: Do we still need/want this here if we have it in the class init now as well?

File: tests/providers/amazon/aws/hooks/test_sqs.py [hunk 2]
```diff
@@ -28,8 +28,8 @@
 
 
 @unittest.skipIf(mock_sqs is None, 'moto sqs package missing')
-class TestAwsSQSHook(unittest.TestCase):
+class TestAwsSqsHook(unittest.TestCase):
```
Comment (ferruzzi) [inline tests/providers/amazon/aws/hooks/test_sqs.py]: Should this be changed to `TestSqsHook`?

**Review by ferruzzi — COMMENTED** [review]
Left a couple of notes.

File: airflow/contrib/operators/aws_sqs_publish_operator.py (hunk 1 above)
Comment (eladkal) [reply airflow/contrib/operators/aws_sqs_publish_operator.py:28]: It's like that for other hooks so I left it as is:
https://github.com/apache/airflow/blob/main/airflow/contrib/hooks/gcp_spanner_hook.py
https://github.com/apache/airflow/blob/main/airflow/contrib/hooks/aws_hook.py

File: tests/providers/amazon/aws/hooks/test_sqs.py (hunk 2 above)
Comment (eladkal) [reply tests/providers/amazon/aws/hooks/test_sqs.py]: changed

**Review by ferruzzi — APPROVED** [review]
LGTM

File: airflow/providers/amazon/aws/operators/sqs.py [hunk 3]
```diff
@@ -89,3 +90,19 @@ def execute(self, context: 'Context'):
...
+class SQSPublishOperator(SqsPublishOperator):
+    """
+    This operator is deprecated.
+    Please use :class:`airflow.providers.amazon.aws.operators.sqs.SqsPublishOperator`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        warnings.warn(
+            "This operator is deprecated. "
+            "Please use :class:`airflow.providers.amazon.aws.operators.sqs.SqsPublishOperator`.",
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/operators/sqs.py]:
```suggestion
            "Please use `airflow.providers.amazon.aws.operators.sqs.SqsPublishOperator`.",
```

Deprecation warnings are not rst-rendered

File: airflow/providers/amazon/aws/sensors/sqs.py [hunk 4]
```diff
@@ -196,3 +197,19 @@ def filter_messages_jsonpath(self, messages):
...
+class SQSSensor(SqsSensor):
+    """
+    This sensor is deprecated.
+    Please use :class:`airflow.providers.amazon.aws.sensors.sqs.SqsSensor`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        warnings.warn(
+            "This class is deprecated. "
+            "Please use :class:`airflow.providers.amazon.aws.sensors.sqs.SqsSensor`.",
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/sensors/sqs.py]:
```suggestion
            "Please use `airflow.providers.amazon.aws.sensors.sqs.SqsSensor`.",
```

**Review by uranusjr — APPROVED** [review]

File: airflow/providers/amazon/aws/sensors/sqs.py (hunk 4 above)
Comment (ferruzzi) [reply airflow/providers/amazon/aws/sensors/sqs.py]: Nice catch, sorry I missed those.

**Pull request merged**